### ISSUE TITLE
nil pointer site.Home.RelPermalink on server reloads

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#7603]
I have seen this some times on server reloads, which indicate some race condition:

```bash
at <site.Home.RelPermalink>: error calling RelPermalink: runtime error: invalid memory address or nil pointer dereference
```

**Comment by LandGod** [comment]
I ran into this (or perhaps a similar but separate issue) a TON today. It seemed that I would often get this error due to providing an invalid string substitution argument to `js.Build` `defines`. I wonder if this is simply an issue of not throwing an error in js.Build, causing the built document to be `nil`, and then we get this nil pointer error. Further bolstering this theory is that sometimes instead of the error, the build would work, but then the JS file just wouldn't be there. 

For further context, I had a dictionary of `defines` values that had previously been working, but when I tried to add the following, I would continuously get the above error (not for `site.Home` of course, but the rest of the message is an exact match):
```
// Tried replacing a part of an existing string with a new value
{{ $mainJsReplace  = merge $mainJsReplace   ( dict "HUGO_GEO_API_URL" .Site.BaseURL ) ) }}
```
After changing it to the following, everything works:
```
// Gave up on trying to replace part of the string, and instead constructed the entire thing so I could do a full replacement.
{{ $mainJsReplace  = merge $mainJsReplace   ( dict "HUGO_GEO_API_URL" (print "\"" .Site.BaseURL ".netlify/functions/geolocate" "\"" ) ) }}
```

So in summary, it seems like perhaps there is a `js.Build` error happening somewhere but it's being ignored, leading to the entire function ultimately returning `nil`. 

I hope this is helpful.

**Comment by regisphilibert** [comment]
I have a project where I encounter the problem on both server and regular build since I upgrade form **0.64.0** to **0.65.0**. But it only comes from the 404 template, if I remove the `404` layout the site builds with no problem.

If this does not help I'll try to identify the difference between this project and another running building its 404 with no issue on **0.69.1**


I have the error from the 404 template on both `site.Home.RelPermalink`:
```
Error building site: render of "404" failed: execute of template failed: template: 404.html:38:29: [...] at <site.Home.Permalink>: error calling Permalink: runtime error: invalid memory address or nil pointer dereference
```

And `site.GetPage`
```
Error building site: render of "404" failed: execute of template failed: template: 404.html:38:29: [...] at <site.GetPage>: error calling GetPage: runtime error: invalid memory address or nil pointer dereference
```

**Comment by Darkproduct** [comment]
I had this error pop up when adding a homepage in `content/_index.md` and changing the date value to the current time. 
```
---
title: Home
type: page
date: 2023-05-05T18:00:00+00:00
---
```
```
❯ date
Sa 6. Mai 00:36:52 CEST 2023
❯ date --utc
Fr 5. Mai 22:36:54 UTC 2023
```

Seems like the date value must be earlier then the current UTC time which makes sense, but I didn't think of this in the moment.

**Comment by bep** [comment]
I'm closing this, haven't seen this after the rewrite, assume fixed.
